### PR TITLE
InvalidInputException: fix "illegally instantiates" warnings

### PR DESCRIPTION
**Pull request by jukzi**
InvalidInputException is marked @noinstantiate and should only be constructed by Scannner

### CODE REVIEW

File: org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/core/compiler/InvalidInputException.java [hunk 1]
```diff
@@ -28,7 +28,7 @@ public class InvalidInputException extends Exception {
  * Creates a new exception with no detail message.
  */
 public InvalidInputException() {
-	super();
+	org.eclipse.jdt.internal.compiler.parser.Scanner.invalidInput();
```
Comment (iloveeclipse) [inline org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/core/compiler/InvalidInputException.java:31]: Not sure I understood why `Scanner.invalidInput()` is added here?

Comment (jukzi) [reply org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/core/compiler/InvalidInputException.java:31]: thanks. left over from refactoring

**Comment by iloveeclipse** [comment]
> InvalidInputException is marked @noinstantiate and should only be constructed by Scannner

I believe this is a misunderstanding. The code was originally in the jdt.core bundle, and `@noinstantiate` was for the "all other bundles" outside JDT core. Now the code moved from jdt core to batch compiler, jdt core itself became "other" client (which is not the case). Instead of this refactoring probably better to silence the warning in jdt code.

**Comment by jukzi** [comment]
> Instead of this refactoring probably better to silence the warning in jdt code.

I prefer solutions (here: moving back the code to jdt core) over disabling warnings, which could be helpful in other circumstances

**Pull request merged**